### ISSUE TITLE
KeyRecursiveObservation.ToKeyValueNodes throws exception for non-leaf nodes

### ISSUE THREAD

**Issue by tevert** [andyalm/consul-rx#4]
https://github.com/andyalm/consul-rx/blob/83e262267040a4e233c931c1bba42d21a31b533f/ConsulRx/KeyRecursiveObservation.cs#L24

When the results includes updates for a non-leaf node/folder node, the value field is null and we get an exception here. I would suggest filtering out the folder nodes from this convenience method.

**Comment by andyalm** [comment]
Can you post the specific exception you are getting (with stack trace)? I'm unable to reproduce at the moment.

**Comment by tevert** [comment]
```
System.ArgumentNullException
  HResult=0x80004003
  Message=Array cannot be null.
Parameter name: bytes
  Source=mscorlib
  StackTrace:
   at System.Text.Encoding.GetString(Byte[] bytes)
   at ConsulRx.KeyRecursiveObservation.<>c.<ToKeyValueNodes>b__9_0(KVPair p)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at TestClient.ConfigObserver.OnNext(KeyRecursiveObservation value) in C:\Users\tre\work\vault-dev\TestClient\TestClient\ConfigObserver.cs:line 21
   at System.Reactive.AutoDetachObserver`1.OnNextCore(T value)
   at System.Reactive.ObserverBase`1.OnNext(T value)
   at ConsulRx.ObservableConsul.<>c__DisplayClass12_0`2.<<LongPoll>b__0>d.MoveNext()
```

Does that help?

**Comment by andyalm** [comment]
I have pushed a fix. I will try to get this pushed to NuGet shortly.

**Comment by tevert** [comment]
Thanks!

**Comment by andyalm** [comment]
This has been released in 1.0.0-rc3. Let me know if you encounter any more issues.
